crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -662,4 +725,210 @@ mod tests {
         world.flush();
         assert_eq!(1, world.resource::<R>().0);
     }
+
+    #[test]
+    fn observer_propagating() {
+        let mut world = World::new();
+        world.init_resource::<R>();
+
+        let parent = world
+            .spawn_empty()
+            .observe(|_: Trigger<EventPropagating>, mut res: ResMut<R>| res.0 += 1)
+            .id();
+
+        let child = world
+            .spawn(Parent(parent))
+            .observe(|_: Trigger<EventPropagating>, mut res: ResMut<R>| res.0 += 1)
+            .id();
+
+        // TODO: ideally this flush is not necessary, but right now observe() returns WorldEntityMut
+        // and therefore does not automatically flush.
+        world.flush();
+        world.trigger_targets(EventPropagating, child);
+        world.flush();
+        assert_eq!(2, world.resource::<R>().0);
+    }
+
+    #[test]
+    fn observer_propagating_redundant_dispatch() {
+        let mut world = World::new();
+        world.init_resource::<R>();
+
+        let parent = world
+            .spawn_empty()
+            .observe(|_: Trigger<EventPropagating>, mut res: ResMut<R>| res.0 += 1)
+            .id();
+
+        let child = world
+            .spawn(Parent(parent))
+            .observe(|_: Trigger<EventPropagating>, mut res: ResMut<R>| res.0 += 1)
+            .id();
+
+        // TODO: ideally this flush is not necessary, but right now observe() returns WorldEntityMut
+        // and therefore does not automatically flush.
+        world.flush();
+        world.trigger_targets(EventPropagating, [child, parent]);
```

Comment:
Could you add a `world.trigger_targets(EventPropagating, [child, child]);` too?

I can't see docs about not being allowed to trigger the same entity twice so I'm curious what happens.
And I think I'd expect 4 on `R`?

Reply:
Will add. Should be allowed, would expect 4 as well.